backend-processing.ts

```diff
@@ -344,10 +392,129 @@ export default class BackendProcessing {
-    return null;
+  getHistoryOfIndividual(individualID): Promise<DisplayHistory[]> {
+    return new Promise<DisplayHistory[]>((resolve, reject) => {
+      database.getIndividualEmployeeHistory(individualID).then(results => {
+        resolve(
+          results.map(entry => {
+            let history: DisplayHistory = {
+              id: entry.id,
+              company: entry.Company.CompanyName,
+              position: entry.PositionName,
+              start: entry.StartDate,
+              end: entry.EndDate
+            };
+            return history;
+          })
+        );
+      });
+    });
+  }
+
+  insertHistory(history, individual, userID): Promise<Boolean> {
+    return new Promise<Boolean>((resolve, reject) => {
+      let fundID = individual.FundID;
+      database
+        .retrieveCompanyByName(history.company, fundID)
+        .then(resultCompany => {
+          database
+            .insertEmployeeHistory(userID, history.id, resultCompany.CompanyID, history.position, history.start, history.end)
+            .then(result => {
+              if (result != null) resolve(true);
+              else resolve(false);
+            })
+            .catch(error => {
+              console.error("An error occurred while inserting employee history");
+              console.error(error);
+              resolve(false);
+            });
+        })
+        .catch(error => {
+          console.error("An error occurred while retrieving company information");
+          console.error(error);
+          resolve(false);
+        });
+    });
+  }
+
+  //The information passed into this function pertains to individuals, employeeHistory (except company ID), and company name
+  //This function assumes that the individualID part of employeeHistory may change, but the details about the individual will not be changed here,
+  //there is a seperate function for that. Similarly, this function assumes that the companyID may change, but the name of the specified company is not changing.
+  updateHistory(history, individual, userID): Promise<Boolean> {
+    return new Promise<Boolean>((resolve, reject) => {
+      let fundID = individual.FundID;
+      database
+        .retrieveCompanyByName(history.company, fundID)
+        .then(resultCompany => {
+          database
+            .updateHistory(
+              history.HistoryID,
+              userID,
+              individual.IndividualID,
+              resultCompany.CompanyID,
+              history.position,
+              history.start,
+              history.end
+            )
+            .then(resolve(true));
+        })
+        .catch(error => {
+          console.error("An error occurred while retrieving company information");
+          console.error(error);
+          resolve(false);
+        });
+    });
+  }
+
+  //returns false if username or email is unavailable
+  handleSignup(username, email, password): Promise<Boolean> {
```

Comment:
We already have a signup function inside the `/signup` api endpoint in `server.ts` class. Is there a reason why you needed this one?

Reply:
There was a miscommunication and we both worked on that part separately. I was planning on deleting my work on that since your code was more concise, evidently I missed part of it.